### DISCUSSION TITLE
Pandoc Plugin Obsidian, can’t find bibliography file

### DISCUSSION THREAD

**Discussion by jackrgilmore** [jgm/pandoc#10182] [Q&A] [answered]
Does anyone know how to work with this?

### What I’m trying to do

Export to various formats using Pandoc to change my citekeys into citations and add a bibliography to my documents. Following advice from [this tutorial](https://betterhumans.pub/obsidian-tutorial-for-academic-writing-87b038060522#cb65) and [this tutorial](https://medium.com/@alexandraphelan/an-updated-academic-workflow-zotero-obsidian-cffef080addd)

I keep getting an error message that says the file or directory doesn’t exist in the resource path

My Extra Pandoc Arguments:

```
--resource-path="/Users/jgilmore/Library/Mobile Documents/iCloud~md~obsidian/Documents/Second-Brain/"
--bibliography="/Users/jgilmore/Documents/Second-Brain/Atlas/Utilities/Zotero/Zotero.bib"
--number-sections 
--citeproc
-M reference-section-title=References
-o output.docx
```

### Things I have tried
- troubleshooting the hell out of it with ChatGPT
- moving my betterbibtex file and betterbibtex CSL json file into my vault and directing it there
- changing various things in the arguments for over 3 hours straight

At this point, I'm very frustrated. I can see the potential this tool has for academic writing so I really want to get it to work. 

> **Note:** The plugins I'm using are Citations, Pandoc Reference List, and Pandoc Plugin

**Comment by jgm** [comment]
I assume
```
ls "/Users/jgilmore/Documents/Second-Brain/Atlas/Utilities/Zotero/Zotero.bib"
```
shows that the file is there?

I would suggest trying a pure pandoc command rather than using the plugin, to see if the bibliography is found.  I don't think this is a pandoc issue.  You may need to consult the maintainers of the plugin (we have nothing to do with that).

**Comment by Jmuccigr** [comment]
I second @jgm’s advice, esp. about using the command line.

For two obsidian things:

1. Remove the quotation marks
2. Put the biblio line into the plug-in’s preferences in the last item for an additional command. That’s where mine is and it works. No quotation marks.

**Comment by jackrgilmore** [comment]
@jgm & @Jmuccigr 

Thank you for your replies. I changed the formatting a bit. I still receive the same error message about the file not existing, even after checking with `ls` in terminal. Here's what it looks like now:

```
--number-sections
--citeproc
-M reference-section-title=References
--bibliography="/Users/jgilmore/Library/Mobile Documents/iCloud~md~obsidian/Documents/Second-Brain/Atlas/Utilities/Zotero/Zotero.bib"
```


And here's the error message I keep receiving:

<img width="1512" alt="Screenshot 2024-09-14 at 10 02 11" src="https://github.com/user-attachments/assets/5736746b-5fd8-4db9-985c-a1d195f854f2">



I've posted on other forums more specific to the plugin, but so far you guys have been the most responsive, so thank you.

**Reply by jgm** [reply]
It looks like you're using the `~` character in the path to your bibliography. Most shells will expand this to the home directory and it looks like that's what is happening.  Put a backslash in front of these occurrences of `~` in the path to escape them.

**Comment by Jmuccigr** [comment, marked as answer]
There's a [known bug](https://github.com/OliverBalfour/obsidian-pandoc/issues/159) where the plugin can't handle biblio (and other) files with a space in their path, which is what you now have in your biblio yaml in the "Mobile Documents" part of it. Note how the error starts with just "Documents" and not the whole file path. Move the file to someplace else and see what happens. (Back where it was is probably a good start.)

I'm also suspicious of the trailing quotation mark in the path in the error. My path doesn't use quotation marks, as I noted already.

Here's what I have this as my "Extra pandoc arguments", the last item in the plugin preferences: `--bibliography=/Users/username/Documents/github/local/miscellaneous/MyLibrary.json --citeproc`

Note the `--citeproc`.

**Reply by jackrgilmore** [reply]
Thank you! That fixed the problem
